### ISSUE TITLE
APO defaultObjectRights.xml needs normalization code

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/dor-services-app#2709]
NOTE:  Do #2797 first.

PR  #2677 has an xit-ed spec on line 421 that shows how license information changes for defaultObjectRights.xml for APO objects.

for an APO, `defaultObjectRights.xml`  might contain

```xml
  <rightsMetadata>
            <access type="discover">
              <machine>
                <world/>
              </machine>
            </access>
            <access type="read">
              <machine>
                <world rule="no-download"/>
              </machine>
            </access>
            <use>
              <human type="useAndReproduction">Use at will.</human>
              <human type="creativeCommons">CC BY-NC Attribution-NonCommercial</human>
              <machine type="creativeCommons">by-nc</machine>
            </use>
</rightsMetadata>
```

but after roundtripping through cocina, it becomes something like this:

```
 <rightsMetadata>
(snip)
            <use>
              <human type="useAndReproduction">Use at will.</human>
              <human type="creativeCommons">https://whatever_the-url-is.org</human>
              <machine type="creativeCommons">https://whatever_the-url-is.org</machine>
            </use>
</rightsMetadata>
```

We should have "normalization" code, such as we do for MODS/descriptive, to ignore this difference when comparing roundtripped Fedora xml (fedora -> cocina -> roundtripped fedora).  (right @andrewjbtw ???)

The normalization could/should be executed by bin/(validate-cocina-roundtrip) scripts as appropriate and it should also be executed by spec/services/cocina/mapping/administrative/apo_administrative_spec.rb so the specs compare the actual APO roundtripped Fedora defaultObjectRights.xml to the defaultObjectRights.xml that is normalized by executing said normalization code.

Note that this may also come into play for rightsMetadata.xml !!!

**Comment by justinlittman** [comment]
I'm unclear. Isn't all that needs to happen here is remove `use/machine[@type='creativeCommons']` and `use/human[@type='creativeCommons']` since the values are not going to be comparable between the original and roundtripped? (Like I do here: https://github.com/sul-dlss/dor-services-app/pull/2760/files#diff-220b639ac3fca24612e65344c19ff0a997a7f23921e1339890d2bdfeba43a7ffR61-R64)

Is there anything else that needs to be normalized?

If so, that normalizer could be used for APO default rights, as well as rightsMetadata normalization.

**Comment by ndushay** [comment]
I think it goes from a sort of code to a URL -- and I assumed the normalization code would do the expected conversion ... and yes, it could be used for rightsMetadata as well as APO default rights.

**Comment by peetucket** [comment]
ok, the refactoring to get the normalization code into an easily extendable class is now complete, so I think we can return to this ticket...but it is still not clear to me what precisely is the next step

I presume the roundtripping is already doing the work of turning licenses from strings into URIs, so the goal is to ensure that when the roundtrip validation code runs, it realizes this is OK and that they should match?  I think I may need a discussion with @justinlittman to be sure

**Comment by ndushay** [comment]
I think #2797 should happen first.   I can help you with your question for 2797.

**Comment by justinlittman** [comment]
closing in favor of #3129

**Comment by jcoyne** [comment]
@justinlittman did you mean to close this issue?
